packages/core/src/validate.test.ts

```diff
@@ -133,3 +133,17 @@ testValid('UsesExplicitSafeExternalLibraryNatspec', 'transparent', true);
 testValid('UsesExplicitUnsafeInternalLibraryNatspec', 'transparent', true);
 testValid('UsesExplicitUnsafeExternalLibraryNatspec', 'transparent', true);
 testValid('TransitiveLibraryIsUnsafe', 'transparent', false);
+
+testValid('contracts/test/ValidationsSameNameSafe.sol:SameName', 'transparent', true);
+testValid('contracts/test/ValidationsSameNameUnsafe.sol:SameName', 'transparent', false);
+
+test('ambiguous name', t => {
+  const getAmbiguousName = () => getContractVersion(t.context.validation, 'SameName');
+  const error = t.throws(getAmbiguousName);
```

Comment:
I'd usually write:

```suggestion
  const error = t.throws(() => getContractVersion(t.context.validation, 'SameName'));
```

Do you have a preference?